includes/Modules/Analytics.php

```diff
@@ -1128,12 +1089,18 @@ function( $property ) use ( $urls ) {
 
 					$result = array_merge( $response, $properties );
 
-					if ( $existing_tag ) {
-						$result = array_merge( $result, array( 'existingTag' => $has_access_to_property ) );
+					// Get matched property from exiting tag property id.
+					if ( ! empty( $existing_tag ) ) {
+						$matched_property = array_filter(
+							$properties['properties'],
+							function( $property ) use ( $existing_tag ) {
+								return $property->getId() === $existing_tag['property'];
+							}
+						);
 					}
 
 					if ( $matched_property ) {
-						$result = array_merge( $result, array( 'matchedProperty' => $matched_property ) );
+						$result = array_merge( $result, array( 'matchedProperty' => array_values( $matched_property ) ) );
```

Comment:
While we're at it, might be good to rename this to `matchedProperties` (the variable in a similar way), since it's an array of properties.

Reply:
@felixarntz I believe only 1 matched property is possible, either it matches URL or existing tag. I suggest returning the actual object to the client side instead of an array of 1 element and keep the naming as matchedProperty, so just returning the first element of `$matched_property`

Reply:
@felixarntz I changed this here https://github.com/google/site-kit-wp/pull/434/commits/b733ba1190b879b286a22c0435a72fdaa4a98343

Reply:
That makes sense, thanks for clarifying!